pkg/tsdb/elasticsearch/client/client_test.go

```diff
@@ -11,310 +11,276 @@ import (
-			Database: "[metrics-]YYYY.MM.DD",
-			JsonData: simplejson.NewFromAny(map[string]interface{}{
-				"esVersion":                  5,
-				"maxConcurrentShardRequests": 100,
-				"timeField":                  "@timestamp",
-				"interval":                   "Daily",
-			}),
-		}, func(sc *scenarioContext) {
-			sc.responseBody = `{
+	httpClientScenario(t, "Given a fake http client and a v5.x client with response", &models.DataSource{
+		Database: "[metrics-]YYYY.MM.DD",
+		JsonData: simplejson.NewFromAny(map[string]interface{}{
+			"esVersion":                  5,
+			"maxConcurrentShardRequests": 100,
+			"timeField":                  "@timestamp",
+			"interval":                   "Daily",
+		}),
+	}, func(sc *scenarioContext) {
+		sc.responseBody = `{
 				"responses": [
 					{
-						"hits": {	"hits": [], "max_score": 0,	"total": 4656	},
+						"hits": { "hits": [], "max_score": 0, "total": 4656 },
 						"status": 200
 					}
 				]
 			}`
 
-			Convey("When executing multi search", func() {
-				ms, err := createMultisearchForTest(sc.client)
-				So(err, ShouldBeNil)
-				res, err := sc.client.ExecuteMultisearch(ms)
-				So(err, ShouldBeNil)
+		ms, err := createMultisearchForTest(t, sc.client)
+		require.NoError(t, err)
+		res, err := sc.client.ExecuteMultisearch(ms)
+		require.NoError(t, err)
 
-				Convey("Should send correct request and payload", func() {
-					So(sc.request, ShouldNotBeNil)
-					So(sc.request.Method, ShouldEqual, http.MethodPost)
-					So(sc.request.URL.Path, ShouldEqual, "/_msearch")
+		require.NotNil(t, sc.request)
+		assert.Equal(t, http.MethodPost, sc.request.Method)
+		assert.Equal(t, "/_msearch", sc.request.URL.Path)
 
-					So(sc.requestBody, ShouldNotBeNil)
+		require.NotNil(t, sc.requestBody)
 
-					headerBytes, err := sc.requestBody.ReadBytes('\n')
-					So(err, ShouldBeNil)
-					bodyBytes := sc.requestBody.Bytes()
+		headerBytes, err := sc.requestBody.ReadBytes('\n')
+		require.NoError(t, err)
+		bodyBytes := sc.requestBody.Bytes()
 
-					jHeader, err := simplejson.NewJson(headerBytes)
-					So(err, ShouldBeNil)
+		jHeader, err := simplejson.NewJson(headerBytes)
+		require.NoError(t, err)
 
-					jBody, err := simplejson.NewJson(bodyBytes)
-					So(err, ShouldBeNil)
+		jBody, err := simplejson.NewJson(bodyBytes)
+		require.NoError(t, err)
 
-					So(jHeader.Get("index").MustString(), ShouldEqual, "metrics-2018.05.15")
-					So(jHeader.Get("ignore_unavailable").MustBool(false), ShouldEqual, true)
-					So(jHeader.Get("search_type").MustString(), ShouldEqual, "query_then_fetch")
-					So(jHeader.Get("max_concurrent_shard_requests").MustInt(10), ShouldEqual, 10)
+		assert.Equal(t, "metrics-2018.05.15", jHeader.Get("index").MustString())
+		assert.True(t, jHeader.Get("ignore_unavailable").MustBool(false))
+		assert.Equal(t, "query_then_fetch", jHeader.Get("search_type").MustString())
+		assert.Empty(t, jHeader.Get("max_concurrent_shard_requests"))
```

Comment:
Previously, the test did `jHeader.Get("max_concurrent_shard_requests").MustInt(10)` which meant that a default value of 10 was returned if no value was provided/set.

Here, `max_concurrent_shard_requests` is set in the request (`"maxConcurrentShardRequests": 100`). Although `max_concurrent_shard_requests` is only set for Elasticsearch version >= 5.6 && < 7 in https://github.com/grafana/grafana/blob/6f72ae38d5b466caf5487696a4247d358bbd298f/pkg/tsdb/elasticsearch/client/client.go#L303-L306, so the expected value should still be empty because this test is for Elasticsearch version 5. 

Let's ensure we assert that we got an empty value.